File: ts-sdk/src/sdk/models/shared/shareduser.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// SharedUser
/** 
 * The resource creator
**/
export class SharedUser extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=github_username" })
  githubUsername?: string;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=profile_image" })
  profileImage?: string;

  @SpeakeasyMetadata({ data: "json, name=profile_image_90" })
  profileImage90?: string;

  @SpeakeasyMetadata({ data: "json, name=twitter_username" })
  twitterUsername?: string;

  @SpeakeasyMetadata({ data: "json, name=username" })
  username?: string;

  @SpeakeasyMetadata({ data: "json, name=website_url" })
  websiteUrl?: string;
}

File: go-sdk/pkg/models/shared/shareduser.go
package shared

// SharedUser
// The resource creator
type SharedUser struct {
	GithubUsername  *string `json:"github_username,omitempty"`
	Name            *string `json:"name,omitempty"`
	ProfileImage    *string `json:"profile_image,omitempty"`
	ProfileImage90  *string `json:"profile_image_90,omitempty"`
	TwitterUsername *string `json:"twitter_username,omitempty"`
	Username        *string `json:"username,omitempty"`
	WebsiteURL      *string `json:"website_url,omitempty"`
}
